app/src/main/java/com/andrew/tasky/presentation/agenda/AgendaFragment.kt

```diff
@@ -1,64 +1,279 @@
-                "Take trash from inside to outside", "4:00", "Mar 5"),
-            AgendaItem(AgendaItemType.TASK, true, "Code",
-                "Finish coding lesson", "5:00", "Mar 5"),
-            AgendaItem(AgendaItemType.EVENT, false, "Run",
-                "Go for an evening run", "6:00", "Mar 5"),
+        viewModel = ViewModelProvider(this)[AgendaViewModel::class.java]
+        fragmentAgendaBinding = FragmentAgendaBinding.bind(view)
+        currentDateAndTime = LocalDateTime.now()
+
+        val agendaItemList = mutableListOf(
+            AgendaItem(
+                AgendaItemType.EVENT, false, "Meeting",
+                "Project meeting with Abby", LocalDateTime.of(2022, 10, 3, 8, 22),
+                LocalDateTime.of(2022, 10, 3, 8, 32)
+            ),
+            AgendaItem(
+                AgendaItemType.REMINDER, false, "Take out trash",
+                "Take trash from inside to outside", LocalDateTime.of(2022, 10, 3, 8, 37)
+            ),
+            AgendaItem(
+                AgendaItemType.TASK, true, "Code",
+                "Finish coding lesson", LocalDateTime.of(2022, 10, 3, 9, 22)
+            ),
+            AgendaItem(
+                AgendaItemType.EVENT, false, "Run",
+                "Go for an evening run", LocalDateTime.of(2022, 10, 3, 12, 22),
+                LocalDateTime.of(2022, 10, 15, 14, 22)
+            ),
         )
 
         val adapter = AgendaItemAdapter(agendaItemList)
 
-        binding.agendaItemRecyclerView.adapter = adapter
-        binding.agendaItemRecyclerView.layoutManager = LinearLayoutManager(view.context)
+        fragmentAgendaBinding.agendaItemRecyclerView.adapter = adapter
+        fragmentAgendaBinding.agendaItemRecyclerView.layoutManager =
+            LinearLayoutManager(view.context)
+
+        subscribeToObservables()
+        setOnClickListeners()
+    }
+
+    private fun <T> Fragment.collectLatestLifecycleFlow(
+        flow: Flow<T>,
+        onCollect: suspend (T) -> Unit
+    ) {
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                flow.collectLatest {
+                    onCollect(it)
+                }
+            }
+        }
+    }
+
+    private fun subscribeToObservables() {
 
-        binding.addTask.setOnClickListener(){
-            val agendaItemType = AgendaItemType.TASK.name
-            navController.navigate(AgendaFragmentDirections
-                .actionAgendaFragmentToAgendaItemDetailFragment(agendaItemType))
+        //miniCalendarSetup function call, passing the currently selected date
+        collectLatestLifecycleFlow(viewModel.dateSelected) { dateSelected ->
+            setupMiniCalendar(dateSelected)
         }
+    }
+
+    private fun setOnClickListeners() {
+
+        //On click listeners for miniCalendar allowing user to quickly change to near future dates
+        fragmentAgendaBinding.apply {
+            today.setOnClickListener {
```

Comment:
Might've been better to dynamically occupy this list using a loop. That way you're more flexible when it comes to changing the amount of days you show and it removes quite a bit of boilerplate code here